### PR TITLE
Logging, Tests.

### PR DESCRIPTION
**Pull request by mutexdeveloper**
### Description of the changes

<details>
<summary>Commits</summary>

[Logging system, Test](https://github.com/japandotorg/LemonCord/commit/3e50e0354c966a00adbad34c11bff06033a6d7d4)
[Unneeded wrapping](https://github.com/japandotorg/LemonCord/commit/80d6bbf0dfd619fb28b6b5b6c83b40229c42b3ad)
</details>
Implemented a basic logging system similar to the `console` library in JavaScript
Removed wrapping which doesn't seem necessary, Please correct me if I'm wrong.
<br><br>

**Sorry for being less active, I will try do some more commits soon. I see you have done a lot since.**

### Have the changes in this PR been tested?
**Yes**

### CODE REVIEW

**Comment by mutexdeveloper** [comment]
Clippy is failing and it doesn't look like anything wrong with my changes.

File: README.md [hunk 1]
```diff
@@ -52,43 +54,35 @@ If you are cross-compiling for macOS using [osxcross](https://github.com/tpoecht
...
+ # To check the currently active rustc compiler version, run
 $ rustc --version
-# To check the currently active rustup toolchain manager, run
-$ rustup --version
-# To check the currently active cargo package manager, run
-$ cargo --version
 ```
 
-Alright, now that we have rust installed, it's time to clone and cd to the repository in your machine using the following commands:
+Alright, now that we have rust installed, it's time to download LemonCord
```
Comment (japandotorg) [inline README.md:79]:
```suggestion
- Alright, now that we have rust installed, it's time to download LemonCord
+ Alright, now that we have rust installed, it's time to download LemonCord:
```

a small nitpick, missing punctuations, even tho it's not extremely necessary

File: README.md [hunk 2]
```diff
@@ -98,24 +92,24 @@ $ cargo run --release
...
+ # Open "lemon-cord.exe"
 ```
 
-After that, open the `lemon-cord.exe` executable from your folder.
-You should be able to use Discord just as you could with any other client.
-
 ## Discord / Support
 
-Join our [Discord Server](https://melonbot.io/support) for further Updates, Announcements, Support or just to chat with the developer.
+Join our [Discord Server](https://melonbot.io/support) for further updates, announcements and Support, or just to chat with the developer.
```
Comment (japandotorg) [inline README.md:106]:
```suggestion
Join our [Discord Server](https://melonbot.io/support) for further updates, announcements and support, or just to chat with the developer.
```

File: src/main.rs [hunk 3]
```diff
@@ -183,23 +183,20 @@ async fn discord() -> Result<()> {
...
     #[cfg(any(target_os = "linux", target_os = "windows"))]
-    let mut _webview: Option<wry::webview::WebView> = {
+    let mut _webview: wry::webview::WebView = {
         #[cfg(target_os = "windows")]
         let user_agent: String = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36".to_string();
         
         #[cfg(target_os = "linux")]
         let user_agent = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36".to_string();
-        
-        Some(
```
Comment (japandotorg) [inline src/main.rs:193]: I'd rather have the Some constructor in there

**Comment by japandotorg** [comment]
Since you've already made a logging implementation, you might wanna remove the `tracing` crate from the project and replace all the other `println!`, `error!`, `panic!`, etc etc

**Comment by mutexdeveloper** [comment]
Alright sure.
Sorry for the non completed PR.

File: src/main.rs (hunk 3 above)
Comment (mutexdeveloper) [reply src/main.rs:193]: Is there a reason for this?
It's just a bit annoying. 
I'll add it back, Sorry.

**Comment by japandotorg** [comment]
No worries, not like I was gonna merge it in an incomplete state anyways.

**Comment by mutexdeveloper** [comment]
I believe that's everything @japandotorg .

**Comment by japandotorg** [comment]
looks good to me, the PR will be merged when the latest requested changes have been implemented and I have gotten the reply to my question up there.

**Comment by mutexdeveloper** [comment]
> looks good to me, the PR will be merged when the latest requested changes have been implemented and I have gotten the reply to my question up there.

Sorry I don't understand what you mean.
Am I missing something because I can't find the question you asked and the changes have been commited.

**Comment by mutexdeveloper** [comment]
> - [Here's the question I asked](https://github.com/japandotorg/LemonCord/pull/12#discussion_r1090961056)
> 
> - [The latest changes requested](https://github.com/japandotorg/LemonCord/pull/12#discussion_r1092538098)

Strange. I get an error message clicking on both links. Can you see these messages on an incognito tab? Maybe its an issue with github mobile?

**Comment by japandotorg** [comment]
97793F86-A399-46FF-8CDE-3010E711FABF
86F15D9A-C2B9-44A0-A8E4-9158BDA90348

Here’s the screenshot for them.

**Comment by mutexdeveloper** [comment]
> 97793F86-A399-46FF-8CDE-3010E711FABF
> 86F15D9A-C2B9-44A0-A8E4-9158BDA90348
> 
> The links work for me perfectly fine, but here’s the screenshot for them.

Interesting, I'll fix them today.

**Comment by mutexdeveloper** [comment]
I thought removing the Rust installation guide and replacing it with the official one would be better as it simplifies the README making it less intimidating and the official one is easier to understand.

**Comment by mutexdeveloper** [comment]
Update? @japandotorg .

**Comment by japandotorg** [comment]
everything looks good to me, however I'd like to suggest that the test you've added should be in it's own file/folder

**Comment by mutexdeveloper** [comment]
> everything looks good to me, however I'd like to suggest that the test you've added should be in it's own file/folder

Done. Think that's it.

**Comment by japandotorg** [comment]
Yep LGTM

**Comment by japandotorg** [comment]
Thanks for the PR!

**Pull request merged**